viewer/packages/cad-model/src/picking/PickingHandler.ts

```diff
@@ -113,7 +116,35 @@ export class PickingHandler {
     }
   }
 
-  public async intersectCadNode(
+  private filterOutOfBoundCadNodes(cadNodes: CadNode[], input: IntersectInput) {
+    // Ensure the ray overlaps any point on the given models bounding box.
+    const raycaster = new THREE.Raycaster();
+    raycaster.setFromCamera(input.normalizedCoords, input.camera);
+    const ray = raycaster.ray;
+
+    const getWorldSpaceNodeBounds = (node: CadNode) =>
+      node.cadModelMetadata.scene.root.subtreeBoundingBox.clone().applyMatrix4(node.cadModelMetadata.modelMatrix);
+
+    // Remove all cadNodes that cannot be hit. Avoid a raycast against them
+    const candidateCadNodes = (
```

Comment:
Concider using local functions for readability. See this as example: https://github.com/cognitedata/reveal/blob/christjt/REV-719/viewer/packages/360-images/src/Image360Facade.ts#L76